### DISCUSSION TITLE
Hugo Twitter ShortCode not compiling properly on server, but works locally

### DISCUSSION THREAD

**Discussion by endingwithali** [vercel/community#883] [Help] [answered]
context
https://www.twitch.tv/videos/1595983663

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out! I personally have not worked with Hugo before, so I'm doing some quick learning and looking for other people to help us. Just wanted you to know this is being investigated!

What version of Hugo do you have installed? And can you confirm the repo+branch with the code that's failing? I would guess it's this [blogredesign](https://github.com/endingwithali/blog.ali.dev/tree/blogredesign) branch, but don't want to operate under a faulty assumption.

_Edit:_ You could try setting up a `HUGO_VERSION` [environment variable](https://vercel.com/docs/concepts/projects/environment-variables) to make the Vercel build use the same version that you are using locally. If this error is caused by the versions not matching, that would solve it.

<img width="941" alt="adding environment variable" src="https://user-images.githubusercontent.com/8247105/191371998-0ac22a32-4716-4265-b494-b6b7eb4e82af.png">

This worked for my test site https://quickstart-2epwvvyzg-amyegan.vercel.app/posts/my-first-post/

**Reply by endingwithali** [reply]
this worked! thank you so much!

**Comment by fu-sen** [comment]
I prefer to build Hugo site with ▲Vercel. 😄
Hugo is updated frequently, so specifying the version in `vercel.json` makes it easier to manage.

```
{
  "build": {
    "env": {
      "HUGO_VERSION": "0.103.1"
    }
  }
}
```

**Reply by amyegan** [reply]
Thanks for sharing this! Very helpful to know other ways of setting these values 😄 

There are also a couple of choices for people who prefer to use Vercel CLI or API. The CLI command [`vercel env`](https://vercel.com/docs/cli/env#) or [`/v9/projects/{idOrName}/env/{id}`](https://vercel.com/docs/rest-api#endpoints/projects/edit-an-environment-variable) API endpoint will also let you update `HUGO_VERSION`
